server/server.go

```diff
@@ -824,22 +823,18 @@ func (s *server) RestartNode(ctx context.Context, req *rpcpb.RestartNodeRequest)
 		nodeInfo.DbDir = filepath.Join(req.GetRootDataDir(), req.Name, "db-dir")
 	}
 
-	defaultConfig := local.GetDefaultFlags()
-
 	buildDir, err := getBuildDir(nodeInfo.ExecPath, nodeInfo.PluginDir)
 	if err != nil {
 		return nil, err
 	}
 
-	nodeConfig.ConfigFile, err = createConfigFileString(
-		defaultConfig,
-		nodeInfo.LogDir,
-		nodeInfo.DbDir,
-		buildDir,
-		nodeInfo.WhitelistedSubnets,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to generate json node config string: %w", err)
+	nodeConfig.Flags[config.LogsDirKey] = nodeInfo.LogDir
+	nodeConfig.Flags[config.DBPathKey] = nodeInfo.DbDir
+	if buildDir != "" {
+		nodeConfig.Flags[config.BuildDirKey] = buildDir
+	}
+	if nodeInfo.WhitelistedSubnets != "" {
+		nodeConfig.Flags[config.WhitelistedSubnetsKey] = nodeInfo.WhitelistedSubnets
```

Comment:
Just wondering: should the restarting node keep apiPort and p2pPort?

Reply:
Yes, it should. If you think it is not doing, pls add this comment also to the restart node migration PR so we can fix the problem there

Reply:
Seems to be addressed here: https://github.com/ava-labs/avalanche-network-runner/pull/196#discussion_r932603987